components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/service/impl/password/PasswordRecoveryManagerImpl.java

```diff
@@ -969,7 +970,7 @@ private PasswordResetCodeDTO validateConfirmationCode(UserAccountRecoveryManager
         return buildPasswordResetCodeDTO(confirmationCode);
     }
 
-    private boolean isRecoveryChannelEnabled(String notificationChannelType, String tenantDomain)
```

Comment:
can't we simply return true if notification channel type is external and treat it as enabled by default? I think that change alone would fix the whole thing.